app/objects/c_report.py

```diff
@@ -60,3 +60,142 @@ def store(self, ram):
         existing.update('hosts', self.hosts)
         existing.update('network_map', self.network_map)
         return existing
+
+    def retrieve_host_by_id(self, id):
+        return self.hosts[id]
+
+    # 
+    # Getters / Setters
+    # 
+
+    def get_host_freebie_abilities(self, id):
+        if id in self.hosts:
+            return self.hosts[id].freebie_abilities
+        return None
+
+    def set_host_freebie_abilities(self, id, abilities):
+        if id in self.hosts:
+            self.hosts[id].freebie_abilities = abilities
+            return True
+        return False
+
+    def get_host_acces(self, id):
+        if id in self.hosts:
+            return self.hosts[id].access
+        return None
+    
+    def set_host_access(self, id, access):
+        if id in self.hosts:
+            self.hosts[id].access = access
+            return True
+        return False
+
+    def get_host_denied_abilities(self, id):
+        if id in self.hosts:
```

Comment:
Broadly applicable across all of these, I'm not a big fan of providing the if statement here given the worst that's going to happen is a KeyError is thrown. I think it would be preferable to just catch that exception at a higher level rather than having to perform an if statement on the output after you call each of these to see if it returned an actual value or None. Especially for the setters that return True or False depending on whether the value is set or not. Just requires you to do an additional if statement after calling the setter to see if it worked or not, where that's kinda why exceptions exist in the first place.